### ISSUE TITLE
Arrow-Keys are not immediately reflected to remote prompt

### ISSUE THREAD

**Issue by GeneralGresi** [calebstewart/pwncat#53]
**Describe the bug**
If you are connected to a "Victim" and enter some commands, pressing the arrow keys "up" or "down" does not immediately show the last entered command. If you then hit enter the last command gets executed as normal and gets displayed.
Similar behaviour applies to the arrow keys "left" and "right" - if you enter a command and hit the left arrow key to correct e.g. a typo, the cursor stays at the same place. If you then enter e.g. a space, the cursor get's updated and the space is inserted at the new position. This makes editing files especially hard.

The error does only appear in "remote" mode, it works fine in "local" mode.
Also tried with different reverse shell methods (python, perl, php, bash, nc) - behaviour is the same.

Using other prompt styles and/or syncing the terminal does not solve the problem either.

**Describe the target system**
Tested on multiple Systems, different targets and different attacker machines
Kali Linux in the newest Version
Ubuntu Linux 20.04 LTS newest Version
Tested with different Terminal Programs and also with PuTTY (SSH) 
Appears both in a python-venv and in a normal installation
Using Python 3.8.5

**To Reproduce**
UP/DOWN
1. Connect to a victim
2. Insert a Command
3. Hit the "UP" arrow key to get to the last command
4. The command is not displayed
5. Hit enter, the command gets executed and is shown

LEFT/RIGHT
1. Connect to a victim
2. Type a command
3. Try to correct a typo (by pressing the left or right arrow keys)
4. The cursor does not get updated immediatly
5. At the next keystroke the cursor will get updated eventually

**Expected behavior**
Arrow Keys should be reflected immediately

Screencaptured the error for visual explanation
[pwncat_arrowkeys.zip](https://github.com/calebstewart/pwncat/files/5196996/pwncat_arrowkeys.zip)

**Comment by calebstewart** [comment]
This was an issue that I simply hadn't noticed. Thanks for the report! It will be fixed when the current working brach is merged back to master, which will be relatively soon.

**Comment by calebstewart** [comment]
For anyone who is interested, I used to have both ends of stdio unbuffered, which caused some other issues. In response, I removed the code which set the stdio streams to unbuffered. This meant that the select loop which processed user input could only read one byte at a time. This was fine in theory since if all of the data wasn't read off the buffer, the select loop should immediately notify of available data again (effectively looping through all available data). Oddly, this wasn't happening.

To fix it, I made only `stdin` unbuffered, and now do a larger read in the select loop to account for multi-byte key-presses (like the arrow keys). I was able to reproduce this report before, and it appears to be fixed for me in the `modules` branch.
